x-pack/plugins/ml/public/services/field_service.ts

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { IndexPattern } from 'ui/index_patterns';
+import { Field, Aggregation, AggId } from '../../common/types/fields';
+import { ml } from '../services/ml_api_service';
+import { Dictionary } from '../../common/types/common';
+
+// the type property is missing from the official IndexPattern interface
+interface IndexPatternWithType extends IndexPattern {
+  type?: string;
+}
+
+class FieldsService {
```

Comment:
I think it would be good to get `job` and `caps` in the name of this service / file somehow. `job_caps_service.ts` perhaps? Would also be helpful to add a comment to say what this service is for (particularly as coming up with a name which concisely explains its function is so hard!).

Reply:
I agree with updating the naming - maybe something explicit like `new_job_capabilities_service` like James suggested? 🤔

Reply:
this service is use to create the fields and aggregations and is only used internally in the `newJobCaps` function/endpoint. So i think the name `FieldsService` is applicable here.